tensorflow/compiler/xla/service/gpu/cudnn_batchnorm_rewriter.cc

```diff
@@ -195,15 +265,57 @@ Status Visitor::HandleBatchNormGrad(HloInstruction* batch_norm) {
 
   std::vector<HloInstruction*> operands(batch_norm->operands().begin(),
                                         batch_norm->operands().end());
+  bool is_convert_inserted = false;
+  if (IsBatchNormFP16(batch_norm)) {
+    HloInstruction* operand_0_convert = batch_norm->mutable_operand(0);
+    operands[0] = add_convert(operand_0_convert, F16);
+    for (auto index = 1; index < operands.size(); index++) {
+      if ((batch_norm->operand(index)->opcode() == HloOpcode::kConvert) &&
+          (batch_norm->operand(index)->operand(0)->shape().element_type() ==
+           F16) &&
+          (ShapeUtil::Compatible(
+              operand_0_convert->shape(),
+              batch_norm->mutable_operand(index)->shape()))) {
+        operands[index] = add_convert(batch_norm->mutable_operand(index), F16);
+      }
+    }
+    is_convert_inserted = true;
```

Comment:
Same comment as above -- you probably want to call this `is_batchnorm_with_fp16_inputs` or something like that.